docs/sources/mimir/operators-guide/architecture/query-sharding/index.md

```diff
@@ -139,6 +139,24 @@ parallel by the query-frontend, multiplying the previously set value of
 `-querier.max-query-parallelism` by
 `-query-frontend.query-sharding-total-shards`.
 
+## Cardinality estimation for query sharding (experimental)
+
+Since a high number of parallel sharded queries increases load on queriers and their dependencies, it is desirable to shard queries only as much as necessary.
```

Comment:
```suggestion
Because a high number of parallel sharded queries increases the load on queriers and their dependencies, only shard queries as needed.
```

Reply:
This doesn't sound quite right to me. It's not the user that actively shards queries, which is why it sounds a bit strange to me to phrase this as a directive to the reader. This is just to point out in general that query sharding involves a tradeoff and should only be done by the software as much as is needed for a given query.

As always, you're the expert and I might be completely off 🤷‍♂️